src/app/vertextool/qgsvertexeditor.cpp

```diff
@@ -350,49 +360,40 @@ void QgsVertexEditor::updateVertexSelection( const QItemSelection &selected, con
   mUpdatingVertexSelection = true;
 
   mSelectedFeature->deselectAllVertices();
-  Q_FOREACH ( const QModelIndex &index, mTableView->selectionModel()->selectedRows() )
+
+  QgsCoordinateTransform t( mLayer->crs(), mCanvas->mapSettings().destinationCrs(), QgsProject::instance() );
+  QgsRectangle *bbox = nullptr;
+  QModelIndexList indexList = selected.indexes();
+  for ( int i = 0; i < indexList.length(); ++i )
   {
-    int vertexIdx = index.row();
+    int vertexIdx = indexList.at( i ).row();
     mSelectedFeature->selectVertex( vertexIdx );
-  }
 
-  //ensure that newly selected vertex is visible in canvas
-  if ( !selected.indexes().isEmpty() )
-  {
-    int newRow = selected.indexes().first().row();
-    zoomToVertex( newRow );
+    // create a bounding box of selected vertices
+    QgsPointXY point( mSelectedFeature->vertexMap().at( vertexIdx )->point() );
+    if ( !bbox )
+      bbox = new QgsRectangle( point, point );
+    else
+      bbox->combineExtentWith( point );
   }
 
-  mUpdatingVertexSelection = false;
-}
-
-void QgsVertexEditor::zoomToVertex( int idx )
-{
-  double x = mSelectedFeature->vertexMap().at( idx )->point().x();
-  double y = mSelectedFeature->vertexMap().at( idx )->point().y();
-  QgsPointXY newCenter( x, y );
-
-  QgsCoordinateTransform t( mLayer->crs(), mCanvas->mapSettings().destinationCrs(), QgsProject::instance() );
-  QgsPointXY tCenter;
-  try
+  //ensure that newly selected vertices are visible in canvas
+  if ( bbox )
   {
-    tCenter = t.transform( newCenter );
-  }
-  catch ( QgsCsException & )
-  {
-    return;
+    try
+    {
+      QgsRectangle transformedBbox = t.transform( *bbox );
+      QgsRectangle canvasExtent = mCanvas->mapSettings().extent();
+      transformedBbox.combineExtentWith( canvasExtent );
+      mCanvas->setExtent( transformedBbox );
+    }
+    catch ( QgsCsException & )
+    {
+    }
```

Comment:
Do we really want to ignore such an exception (without printing a message I mean)?